Wraps .table-scroll-wrapper in .form-group class

nao360 opened the pull request:
Provides bottom margin of 30px. Elements rendered after tables are spaced correctly below the table.

![image](https://user-images.githubusercontent.com/6898065/29332041-8916b224-81f6-11e7-8fe3-39a58df8ecaf.png)

stemount commented:
@nao360 check out ParFlowTransitionAdviceListController.php for an alternative.

`    // Show the documents in table format.
    $build['documentation_list'] = [
      '#theme' => 'table',
      '#attributes' => ['class' => 'form-group'],
      '#title' => 'Advice documentation',
      '#header' => ['Document', 'Type of document and regulatory functions', 'Actions', 'Confirmed'],
      '#empty' => $this->t("There is no documentation for this partnership."),
    ];`

nao360 commented:
@stemount - cheers buddy, I'm thinking the wrapper in the markup is advantageous given that we'll always need that margin under tables. Happy to pass it in as an attribute from the controller if you prefer, but I was thinking this way there's one less parameter we've got to pass in.

nao360 commented:
@kalpaitch / @stemount  ParFlowTransitionInspectionPlanForm doesn't have a controller (I suspect because there's only ever one inspection plan to list). This will have to be one for tomorrow, there's no way I'm venturing into refactor-land without a fresh pair of eyes! 

Let's discuss - I still think there's a case to be made for changing the template - from my POV the template is missing the .form-group wrapper to begin with. Alternative would be to extend .table-scroll-wrapper SASS, and give it the bottom margin property it should have.

kalpaitch commented:
So adding .form-group to the table template assumes that all tables are forms... there are a number of cases where this isn't the case.

nao360 commented:
Reverted table template changes (no more .form-group div wrapper), and updated form controller to pass .form-group class via #attributes property.

![image](https://user-images.githubusercontent.com/6898065/29358034-92267e3c-8271-11e7-96a5-e606e3d5a62d.png)

stemount commented on web/modules/custom/par_flow_transition_partnership_details/src/Controller/ParFlowTransitionAdviceListController.php:
```diff
@@ -49,7 +49,7 @@ public function content(ParDataPartnership $par_data_partnership = NULL) {
     $build['documentation_list'] = [
       '#theme' => 'table',
-      '#attributes' => ['class' => 'form-group'],
+      '#attributes' => ['class' => array('form-group')],
```
just to keep things uniform - use [] instead of array()

'#attributes' => ['class' => ['form-group']],

stemount commented on web/modules/custom/par_flow_transition_partnership_details/src/Form/ParFlowTransitionInspectionPlanForm.php:
```diff
@@ -54,7 +54,8 @@ public function buildForm(array $form, FormStateInterface $form_state, ParDataPa
     $form['document_list'] = [
-      '#type' => 'table',
+      '#theme' => 'table',
+      '#attributes' => ['class' => array('form-group')],
```
just to keep things uniform - use [] instead of array()

'#attributes' => ['class' => ['form-group']],

nao360 replied:
Gaaaaaaaaaaaaaaadddddamnit!

stemount commented on web/themes/custom/par_theme/templates/table.html.twig:
```diff
@@ -102,4 +102,4 @@
     {% endif %}
   </table>
-</div>
+</div>
```
no end of line.

nao360 replied:
Curious, there's a clear EOL in mine - just added a second one and pushed the commit. Maybe my git CR/LF config?

kalpaitch commented:
Corrected those @stemount
